tests/load-tests.php

```diff
@@ -148,6 +150,52 @@ public function test_perflab_render_generator() {
 		$this->assertContains( $expected, $output );
 	}
 
+	public function test_perflab_get_valid_modules() {
+		global $wp_version;
+
+		// Assert that it returns the empty array for dummy modules.
+		$dummy_active_modules = array(
+			'../tests/testdata/demo-modules/javascript/demo-module-1',
+			'../tests/testdata/demo-modules/something/demo-module-2',
+			'../tests/testdata/demo-modules/images/demo-module-3',
+		);
+
+		$output = perflab_get_valid_modules( $dummy_active_modules );
+		$this->assertIsArray( $output );
+		$this->assertSame( array( '../tests/testdata/demo-modules/something/demo-module-2', '../tests/testdata/demo-modules/images/demo-module-3' ), $output );
+
+		// Assert that it returns the array for modules.
+		if ( $wp_version >= '6.1' ) {
+			$active_modules = require plugin_dir_path( PERFLAB_MAIN_FILE ) . 'default-enabled-modules.php';
+			add_filter(
+				'perflab_active_modules',
+				function() use ( $active_modules ) {
+					return $active_modules;
+				}
+			);
+			$output = perflab_get_valid_modules( perflab_get_active_modules() );
+
+			// Remove Image module as it will marge in 6.1.
+			$remove_marge_modules = array_shift( $active_modules );
+			$this->assertSame( $active_modules, $output );
+		}
+	}
+
+	public function test_perflab_can_load_module() {
+		// Assert that it validates the ability to load modules that are not in the core.
+		$demo_modules = array(
+			'javascript/demo-module-1' => false,
+			'something/demo-module-2'  => true,
+			'images/demo-module-3'     => true,
+		);
+
+		foreach ( $demo_modules as $module => $can_load ) {
+			$output = perflab_can_load_module( '../tests/testdata/demo-modules/' . $module );
+			$this->assertSame( $can_load, $output );
+		}
```

Comment:
Nit-pick, but this is essentially a `foreach` loop running the same test for different sets of data. For this purpose, a data provider method should be used, and then the test method body should be only what you have here within the `foreach`.

You can set that up via `@dataProvider` annotations. There are several existing data provider usages in the plugin tests or in WordPress core, so you can probably look at those for reference.

Reply:
Thanks for the suggestion for `@dataProvider` annotations. I have added it and updated the test accordingly.